drivers/include/periph/spi.h

```diff
@@ -172,16 +183,18 @@ typedef enum {
  * The actual speed of the bus can vary to some extend, as the combination of
  * CPU clock and available prescaler values on certain platforms may not make
  * the exact values possible.
+ *
+ * @deprecated  Use numeric values instead
+ * @{
  */
-#ifndef HAVE_SPI_CLK_T
-typedef enum {
-    SPI_CLK_100KHZ = 0,     /**< drive the SPI bus with 100KHz */
-    SPI_CLK_400KHZ,         /**< drive the SPI bus with 400KHz */
-    SPI_CLK_1MHZ,           /**< drive the SPI bus with 1MHz */
-    SPI_CLK_5MHZ,           /**< drive the SPI bus with 5MHz */
-    SPI_CLK_10MHZ           /**< drive the SPI bus with 10MHz */
-} spi_clk_t;
-#endif
+enum {
+    SPI_CLK_100KHZ  = KHZ(100), /**< drive the SPI bus with 100KHz */
```

Comment:
Btw: We could implement all backends for now so that `spi_clk_t` will always hold the correct clock divider for the first SPI peripheral and convert it as needed if `spi_acquire()` is called on a bus that is clocked differently.

This would leave some potential for optimization unused, but it would be fully backward compatible. Once the deprecated frequency constants are fully phased out, this optimization potential could be used. Non-breaking API changes are always easier to sell :-)